rbc/omnisci_backend/omnisci_column.py

```diff
@@ -48,77 +33,27 @@ class OutputColumn(Column):
     pass
 
 
-@datamodel.register_default(ColumnPointer)
-class ColumnPointerModel(BufferPointerModel):
-    pass
-
-
-def column_type_converter(obj):
-    """Return Type instance corresponding to Omnisci :code:`Column<T>` type.
-
-    :code:`Omnisci Column<T>` is defined as follows (using C++ syntax)::
-
-      template<typename T>
-      struct Column {
-        T* ptr;
-        size_t sz;
-      }
-
-    See :code:`buffer_type_converter` for details.
-    """
-    return buffer_type_converter(
-        obj, OmnisciColumnType, 'Column', ColumnPointer,
-        extra_members=[])
-
-
-def output_column_type_converter(obj):
-    """Return Type instance corresponding to Omnisci :code:`OutputColumn<T>` type.
-
-    See :code:`column_type_converter` for implementation detail.
-    """
-    return buffer_type_converter(
-        obj, OmnisciOutputColumnType, 'OutputColumn',
-        ColumnPointer, extra_members=[])
-
-
-def table_function_sizer_type_converter(obj):
-    """Return Type instance corresponding to sizer argument of a
-    user-defined table function.
-    """
-    if not isinstance(obj, typesystem.Type):
-        raise NotImplementedError(type(obj))
-    if obj.is_atomic:
-        sizer_name = obj[0]
-        sizer_types = ['RowMultiplier', 'ConstantParameter', 'Constant']
-        if sizer_name in sizer_types:
-            return typesystem.Type.fromstring(f'int32|sizer={sizer_name}')
-
-
 class OmnisciCursorType(typesystem.Type):
 
+    @classmethod
+    def preprocess_args(cls, args):
+        assert len(args) == 1
+        params = []
+        for p in args[0]:
+            if not isinstance(p, OmnisciColumnType):
+                p = OmnisciColumnType((p,))
+            params.append(p)
+        return (tuple(params),)
+
     @property
     def as_consumed_args(self):
-        return list(self)
+        return self[0]
```

Comment:
`self[0]` is a tuple of types, right? i.e. `(OmnisciColumnType(...), OmnisciColumnType(...)`. Can you add a type annotation to it?





```suggestion
@property
def as_consumed_args(self) -> Tuple[typesystem.Type, ...]:
    return self[0]
```

Reply:
You do not need to do this in this PR but at some point we should follow numba and start annotating the codebase

Reply:
I agree we should use annotation to make code easier to read.
However, this PR implements nontrivial features, and to keep diff size of relevant implementation detail small, let's introduce annotations in separate PRs with no new features.